### ISSUE TITLE
blocks disappearing (not visible) 

### ISSUE THREAD

**Issue by Eaglexxv** [ChiselsAndBits/Chisels-and-Bits#118]
AlgorithmX2 
Thank you for your hard work and time great Mods. 
I'm using C&B 11.9 beta for this this is not happening in 11.7 as far as I can tell so far.
I'm using both Chisels&bits and Flat color blocks I have made a few new blocks using the mods together like a lantern also a glowing stone brick for pathways. But now I'm having a problem with the chiseled blocks visually disappearing but still "physically" there . I'm not sure how to show this to you but ill try to do some screen shots. I seems it might have something to do with the glowing blocks. This is more for your info then a complaint. If I can do anything to help you let me know.
1st pic no made glow block everything is fine
2nd is with a made glow block in a different chuck everthing is fine
3rd is with made glow block in same chuck as the house and now any C&B block is visually gone. It still acts like a block is there including the glow. 
Thanks 
Eaglexxv
![2016-08-06_12 42 32](https://cloud.githubusercontent.com/assets/20761687/17458045/2e2f4756-5bd5-11e6-9481-ea1ef1fdcfb4.png)
![2016-08-06_12 42 42](https://cloud.githubusercontent.com/assets/20761687/17458046/32619b80-5bd5-11e6-89e8-40d492517a7f.png)
![2016-08-06_12 43 
[ ] 50](https://cloud.githubusercontent.com/assets/20761687/17458047/368307e4-5bd5-11e6-8875-620d7bdea6a6.png)

**Comment by AlgorithmX2** [comment]
I'd like you to try a few things to see about ironing out what the issue is, please do the first before the second.
1. Could you take a screenshot of your F3 screen with all the details, and post that.
2. Could you try going into the mod settings screen and under "Client Performance" switch "Use VBO" to "NO" instead of "AUTOMATIC" and see if that changes anything.

**Comment by Eaglexxv** [comment]
The VBO did not help, sry. It did help on reload but has soon as I added more blocks I lost the again.

1st is info before adding made blocks w/ VBO automatic

![2016-08-06_21 27 34](https://cloud.githubusercontent.com/assets/20761687/17460016/20d3f826-5c21-11e6-9096-d16869f0da2e.png)

2nd is info after adding made blocks w/ VBO automatic
![2016-08-06_21 27 48](https://cloud.githubusercontent.com/assets/20761687/17460017/3bed61ec-5c21-11e6-8b65-630b7c358299.png)

3rd is info after adding made blocks and VBO no ... after reload

![2016-08-06_21 38 23](https://cloud.githubusercontent.com/assets/20761687/17460020/527be7b2-5c21-11e6-9080-5ba3734e36d2.png)

4th is info after adding more made blocks and VBO no 
![2016-08-06_21 39 32](https://cloud.githubusercontent.com/assets/20761687/17460021/58b0aa00-5c21-11e6-9f91-1a261e6842e6.png)

Hope this helps
Eaglexxv

**Comment by AlgorithmX2** [comment]
Ok, so we learned a few things,
1. That your game was not using VBO's anyway, so as it turns out that setting didn't do anything.
2. Your blocks were visible when they were using the static rendered, and are not visible when using the dynamic renderer.

I've tested this in my world based on your description, and I've had no luck reproducing it.

So here is a new list of things to try and see whats going on.
1. in the same menu as above select "Disable Static Renderer" as true.

This will force everything to use the dynamic one, that one that seems to be having issues. With this setting on compare your results with and without the glowing blocks, and try placing C&B blocks somewhere else.

Heres a list of the questions I guess we'll want answers to after changing this setting,
1. Visible without glowing on building?
2. Visible with glowing on building?
3. Visible without glowing in other chunk.
4. Visible with glowing in other chunk.

If it turns out that every one of these is no, try somewhere more remote, maybe the nether, or a reasonable distance away from the house ( far enough that its out of view distance ). Another world, or the nether would be preferable however. And once again, the same two questions, with and without glowing.

Hopefully this isn't to too painful.

**Comment by Eaglexxv** [comment]
Not painful glad to help

It'll be a few days thou will get back to you

**Comment by AlgorithmX2** [comment]
Alright, just let me know the results when you get around to testing things.

**Comment by weeryan17** [comment]
I also had this problem but switching disable static render to true seams to have fixed it as far as i can tell. I'll so some more scouting around to see if things are fixed for me.

Edit: Nvm Still have the same problem but before i only had one block disappearing in a chunk now it's every chisel and bit block in a chunk

**Comment by AlgorithmX2** [comment]
@weeryan17 In "A Chunk" or in "Any Chunk" aka when you changed the setting to disable the static renderer did all of your chiseled blocks disappear or it it localized to a small area?

If its localized to a small area its quite possible that your have something near by causing it to fail rendering the chiseled blocks. With the Dynamic Renderer.

**Comment by weeryan17** [comment]
@AlgorithmX2 so far i haven't been able to replicate it with any chunks again. So i can tell you for sure it has some sort of randomness to if it works or doesn't. I haven't done much playing so far so this might change an I'll update you if anything happens again.

**Comment by weeryan17** [comment]
@AlgorithmX2 Any chunk it seams it happens when the client is lagging and then skips the chisel and bits rendering for some reason. It also doesn't throw any errors.

**Comment by AlgorithmX2** [comment]
@weeryan17 are you seeing the same behavior with and without the static renderer enabled?

**Comment by weeryan17** [comment]
@AlgorithmX2 no if i have static render enabled it only affects a block instead of a chunk

**Comment by AlgorithmX2** [comment]
@weeryan17 Alright well that at least isolates the dynamic renderer as being key to the problem, if you can try https://dvs1.progwml6.com/jenkins/job/Chisels-and-Bits/lastSuccessfulBuild/artifact/build/libs/chiselsandbits-11.0.8.jar There are some fixes for the dynamic renderer included and its possible things may have changed.

**Comment by weeryan17** [comment]
@AlgorithmX2 It looks like that might have fixed it, but i haven't done much testing yet. I'll get back to you if things break again but it looks like things are working now.

**Comment by AlgorithmX2** [comment]
Ok, let me know how it goes.

**Comment by weeryan17** [comment]
@AlgorithmX2 and it broke again.
Edit: I also learned any bits placed in the chunk also don't appear.

**Comment by Thepacks2** [comment]
@AlgorithmX2 
Greetings from Germany 

I had the same problem. After I installed Optifine, all Chisel and Bits blocks stay visible.
Hopefully this is an information which helps you to fix the bug.

(Please excuse my worse English.)

**Comment by AlgorithmX2** [comment]
Unfortunately that doesn't help, as C&B disables the dynamic renderer when Optifne is installed ( it outright breaks the entire renderer. )

**Comment by weeryan17** [comment]
@AlgorithmX2 when in doubt add debug messages? maybe if you add debug messages to the render part of your code and then send me that version. I get it to be weird again and then send you my log. That's the only idea i have right now

**Comment by AlgorithmX2** [comment]
@weeryan17 Problem is where and what to log? At this point our only real lead is that the issue seems like it is tied to the dynamic renderer, and possibly that lag might cause it to appear.

Just trying to log "everything" will just result in a 2 gb log file ( we are talking a bout rendering ) that isn't helpful, so before we can even try that approach we need more information on what the core problem might be.

**Comment by weeryan17** [comment]
@AlgorithmX2 from what i found it was something to do when a chunk loads sometime chisel and bits gets skipped. and then all current chisel and bits blocks and any placed later don't appear.

**Comment by AlgorithmX2** [comment]
Which is rather unusual, since when you place or changea  chiseled block it should force a re-render of the "dynamic render region" and for some reason in the case of this bug, it can't, or maybe it gets stuck some how...

Here is a question, does the problem affect only those chunks that failed to "render" on load? or all chunks after one that "failed on load"?

aka, if you try placing blocks after it happens in a second chunk or further away are those broken as well?

**Comment by weeryan17** [comment]
@AlgorithmX2 only the failed to render chunks

**Comment by Eaglexxv** [comment]
AlgorithmX2
Sry I am back, I have been following you and weeryan17 (thanks for joining my hell, jk lol) 
I try what you asked , to answer the 4 questions 1&2 are no 3&4 are no and yes it depends how many C&B blocks are in the chuck.  It seems not to matter if it is glowing or not. 
If i go far away and place enough ( seems the number can very as far as how many) it happens in that chuck too.

**Comment by AlgorithmX2** [comment]
Thanks, so everything suggests that there is only one real bug here, and its with the dynamic renderer producing no results.. and getting hung that way for some reason.

The workaround for this would be to set "Minimum Faces for Dynamic Renderer" to an extremely large value, say 90,000"

I'll need to try and do more testing to see if I can reproduce it, any suggestions for a good way to do that, so far I haven't had much luck?

**Comment by Eaglexxv** [comment]
Huummmm ok try sending you my saved world?

**Comment by AlgorithmX2** [comment]
If that building produces the issue reliably that might be a good option.

**Comment by Eaglexxv** [comment]
ok so the questions is how do we do that lol

**Comment by AlgorithmX2** [comment]
Do you have Dropbox or a similar service? If not possibly sign up for one.

**Comment by Eaglexxv** [comment]
I have OneDrive let me work on it give me a few

**Comment by weeryan17** [comment]
@AlgorithmX2 Would you like me to send you my mod pack and world as well?

**Comment by weeryan17** [comment]
Also I think it has something to do with slower computers so if you could somehow slow your minecraft down that would probably help with reproduction.

**Comment by AlgorithmX2** [comment]
@weeryan17 from your description it seemed like your setup wasn't triggering the conditions constantly,  where as @Eaglexxv's world seemed to be more reliable on that front.

So for now I'll see about @Eaglexxv's World first, preferably the issue will only require C&B to test, as extra mods complicate the debugging progress in the IDE.

Inducing lag/stalls is another idea, I'll keep that in mind if the issue continues to prove elusive.

**Comment by Eaglexxv** [comment]
https://1drv.ms/f/s!At3HadxEK73TnkgP3AVQPit-j6RS

**Comment by Eaglexxv** [comment]
let me know when you have it and if loads for you. 
you will getting a warning for a bunch of mods, but I have uninstall all but chuck edge indicator.

**Comment by weeryan17** [comment]
just for your info in case this helps you at all. from what i see it only happens on world load. not on chunk reload/load.

**Comment by weeryan17** [comment]
Just for your info. it also seams it doesn't fix it on a f3+a

**Comment by AlgorithmX2** [comment]
That world was significantly larger then I was expecting. That said I finished downloading it. It loaded fine.

Unfortunately I've still had no luck reproducing the issue in question ( I found a totally unrelated on however, light updates where not re-rendering... )

I've tried introducing delays to the packets and rendering pipelines to try and aggravate things, however that has proved fruitless as well.

I've pretty much run out of ideas at this point.

You both can reproduce this issue with just C&B + Forge, just to confirm that there isn't some strange cross-mod interaction going on here.

**Comment by weeryan17** [comment]
@AlgorithmX2 ok will do. it will take me a while to set up just because i don't want to ruin my current world. I'll also try to see if i can get video of it for you.

**Comment by AlgorithmX2** [comment]
https://dvs1.progwml6.com/jenkins/job/Chisels-and-Bits/lastSuccessfulBuild/artifact/build/libs/chiselsandbits-11.0.9.jar fixes the lighting issue I found, but I doubt it will do much for the visibility problem, but we might as well be testing with the same version.

Thanks, let me know what you find out, I'm not sure a video will help, but I'll take anything at this point.

**Comment by weeryan17** [comment]
@AlgorithmX2 so when i went to just having chisel and bits installed it went back to affecting blocks instead of chunks....... so their still is a problem. Also interesting things happened in testing. i apparently can place bits in the missing block then when i remove it the block updates with the actual chisel design it should be

**Comment by AlgorithmX2** [comment]
That's strangely inconsistent... Which version of forge are you using? I'll try switching to that version.

**Comment by weeryan17** [comment]
Yes yes it is I'm working on the video now

**Comment by weeryan17** [comment]
[video](https://www.youtube.com/watch?v=5ivXI2_AyHA&feature=youtu.be)
Ignore the messed up world. That's because i made a copy of my main world because i thought it would be better. I was right. Broke on 3rd try. Also chunks are hardly loading because i made sure my game was lagging by turning all the settings to max.
Also I'm using forge version 12.18.1.2045

**Comment by AlgorithmX2** [comment]
Hmm, the issue you displayed in the video appears to be an issue with the block itself failing to load from the server, that isn't simply a rendering glitch.

That's why you can try to place stone block inside of it, if it was simply a rendering issue placing the block would yield a stone block above the floor ( the part that you can't walk inside of ) And the fact that when you try to go into it shows that it is de-synced from the server thread ( since it keeps trying to kick you out. )

**Comment by weeryan17** [comment]
@AlgorithmX2 I have no clue. I let you work things out if you need me for any more testing just ask.

**Comment by Eaglexxv** [comment]
Just watch the video my problem is more like you explain @AlgorithmX2 

@weeryan17 can you tell me how you did the video ?

my forge ver is 12.18.1.2046

**Comment by weeryan17** [comment]
@Eaglexxv I used a program called fraps. it isn't free so if your looking for a free way to make a video i suggest obs (Open broadcasting software) or bandicam. bandicam you have to deal with a water mark if you don't pay for it but for something like this it should be fine. Hope in all this chaos we can solve this problem.

**Comment by AlgorithmX2** [comment]
@weeryan17 I'm figuring that your original issue you were first describing wasn't the issue you demonstrated in the video?

Or was it the same? Where the blocks didn't seem to exist client side.

Based on what we saw in the video it seems like MC just didn't bother to send the data, but vanilla is responsible for that, so if anything it seems like that was some kind of vanilla bug manifesting under abnormally stressful conditions... I can't be 100% sure but are your experiencing that under normal conditions?

@Eaglexxv Where you able to try a copy of your world with only C&B and Forge?

**Comment by weeryan17** [comment]
@AlgorithmX2 weird stuff is happening so I have no clue what is part of what bug here. The big problem was I did reload all of my chunks witch should of fixed the issue but as you saw in the video it didn't. so the issue isn't purely vanilla unless vanilla randomly didn't send the data multiple times when it happened with other blocks. I think the chunk not loading problem is happening because of other mods because it stop when it was just chisel and bits installed. And somehow other mods don't like just chisel and bits. I think the individual blocks is a chisel and bits problem but I have no clue beyond it has a high chance of it being server-side

**Comment by AlgorithmX2** [comment]
@weeryan17 So your seeing that missing block data issue commonly? Is it specific to certain blocks? Or entirely random? Your seeing that with a normal view distance setting?

---

I was finally able to reproduce the render issue, I threw Eaglexxv 's world in a 1.10 instance with other mods, I could then reproduce the issue.

Upon removing all the mods from that instance the issue stop reproducing however. This means that there is defiantly some cross mod interactions going on here.

At least I can test now however, There's probably a state being set by a second mod, that is causing C&B to be invisible. I wasn't really expecting that, but it seems to be the case none the less, the only issue now if finding out what that state is.

**Comment by weeryan17** [comment]
@AlgorithmX2 It's not common and it is random. my view distance during that test was set to max but I've had block specific stuff with static rendering on happen with a render distance of 5. also glad you where finally able to reproduce the chunk issue :)

**Comment by weeryan17** [comment]
I'm going to work on the chunk related issue some more by compiling a list of all the mod blocks within a "bad" chunk. If you manage to find the exact problem before i finish the list then yay! Hopefully this list will provide some info to help you.

**Comment by AlgorithmX2** [comment]
I was able to reproduce the render issue with just journeymap and C&B.

**Comment by weeryan17** [comment]
@AlgorithmX2 hmm it must be multiple then because i don't have journey map. slow progress is slow

**Comment by AlgorithmX2** [comment]
Its quite possible, i'm not sure what state is causing the issue yet, but once I track that down it will probably resolve for all of the mods using it.

**Comment by weeryan17** [comment]
Currently all of the chunks that were "bad" have contained tinkers construct stuff their so far hasn't been any other overlap in all the chunks affected. so a state in tinkers construct as well?

**Comment by AlgorithmX2** [comment]
Hard to say for certain, since Journeymap for instance has no blocks, but it renders gui elements.

Once I work out whats going on with Journeymap we'll re-test to make sure its fixed across the board.

**Comment by Eaglexxv** [comment]
I did have tinkers construct also

**Comment by AlgorithmX2** [comment]
So, I'm pretty sure I just chased my tail for like an hour, and I have no idea why I couldn't  reproduce this without a second mod, since when I finally tracked down the issue it was actually a rather simple issue, one that had nothing to do with OpenGL or states, nor any other mod...

Without further ado, hopefully this build fixes the issue, please confirm,
https://dvs1.progwml6.com/jenkins/job/Chisels-and-Bits/10/artifact/build/libs/chiselsandbits-11.0.10.jar

**Comment by Eaglexxv** [comment]
BY JOLLY I THINK YOU HAVE IT 8)!!!!!!!!!!  
Great job looking great been messing around with it for a few hours with everything is stable.
So I confirm that on my side it is fixed.
Hope I was more of a help then a pain in the ass ( but I do like being a pain in the ass) 

if I asked what was the problem It seemed so wierd?

If I can help you in the future let me know
Eaglexxv

**Comment by weeryan17** [comment]
I just woke up so i haven't gotten to testing yet but by the sounds of it everything is fine (besides the server not sending chisel and bits blocks sometimes witch I'll probably make a new issue for).

**Comment by weeryan17** [comment]
Looks like it's fixed. ether that or I'm getting really unlucky making it happen again but i think it's fixed

**Comment by AlgorithmX2** [comment]
@Eaglexxv the issue was that when I was adding VBO support I had to changes  few things, one of those things ended up not getting a proper implementation.

So instead of a list with the block in it, it was just an empty list.

---

Thanks for the help @Eaglexxv, @weeryan17 I'm glad we finally got this fixed, I'll be releasing a new official build later today when I close this issue.

**Comment by Dushess** [comment]
Hello, i have the same bug on 1.9, mod has been updated and after 10.19 (and 11.10) problem is solved. But i building a project using mods, that now released only on 1.9.4 max (or even 1.9). It possible to fix problem for version 9.21? Thanks.
Add: more info about problem manifestation here - [http://www.minecraftforum.net/forums/support/modded-client-support/2752918-wrong-chunks-reloading-ingame-chisels-bits-mod](url)

**Comment by AlgorithmX2** [comment]
@Dushess 
The bug in for this issue was resolved in https://minecraft.curseforge.com/projects/chisels-bits/files/2322263 which is a 1.9.4 build.

However the issue linked above ( https://github.com/AlgorithmX2/Chisels-and-Bits/issues/123 )
was another issue where certain blocks would fail to render, however the fix for that issue requires 1.10.

Its likely that if your already using v10.19 that your issues is actually #123 which requires 1.10
I can't back port that fix, since it requires changes not made in vanilla until 1.10.

**Comment by Dushess** [comment]
Now i use MC1.9 but i using client only mode. (that's not server, my project specified for client side)

**Comment by Dushess** [comment]
I will try 1.9.4 version with other mods on that, and will notify if i'll catch that problem.

**Comment by weeryan17** [comment]
@Dushess do you have a problem where a whole chunk isn't rendering or where only a few blocks are not rendering. if it's a hole chunk the it's probably this just not quite solved all the way. if it's individual blocks the your problem is covered in issue #123 with requires 1.10 to fix.

**Comment by AlgorithmX2** [comment]
@weeryan17  based on the description ( http://www.minecraftforum.net/forums/support/modded-client-support/2752918-wrong-chunks-reloading-ingame-chisels-bits-mod ) the bug appears to affect 1-2 blocks when a chunk loads.

Which sounds quite like the other issue, and not this one.

**Comment by Dushess** [comment]
Add more info about it from #123

> bits can be placed in the block.
In my installation chiseled block in this case just appears again (block state update) and not duplicates.

**Comment by Dushess** [comment]
Upd Info:
Installed same mod set for 1.9.4 - at this moment all fine (knoks the table x3), no chiseled block rendering glitches, no 

> [Client thread/FATAL]: Error executing task
> java.util.concurrent.ExecutionException: java.lang.StackOverflowError
> at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_40]
> at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_40]
> at net.minecraft.util.Util.func_181617_a(SourceFile:46) [g.class:?]
> at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1041) [bcf.class:?]
> at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:366) [bcf.class:?]
> at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_40]
> at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_40]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_40]
> at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_40]
> at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
> at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
> Caused by: java.lang.StackOverflowError
> at net.minecraft.world.World.func_175624_G(World.java:3147) ~[aht.class:?]
> at net.minecraft.world.chunk.Chunk.func_186032_a(Chunk.java:466) ~[ase.class:?]
> at net.minecraft.world.chunk.Chunk.func_177435_g(Chunk.java:461) ~[ase.class:?]
> at net.minecraft.world.chunk.Chunk.func_177422_i(Chunk.java:756) ~[ase.class:?]
> at net.minecraft.world.chunk.Chunk.func_177424_a(Chunk.java:775) ~[ase.class:?]
> at net.minecraft.world.World.func_175625_s(World.java:2380) ~[aht.class:?]
> at mod.chiselsandbits.chiseledblock.BlockChiseled.getTileEntity(BlockChiseled.java:286) ~[BlockChiseled.class:?]
> at mod.chiselsandbits.chiseledblock.BlockChiseled.isNormalCube(BlockChiseled.java:165) ~[BlockChiseled.class:?]
> at net.minecraft.world.World.func_175666_e(World.java:3713) ~[aht.class:?]
> at net.minecraft.world.World.func_175690_a(World.java:2430) ~[aht.class:?]
> at net.minecraft.world.chunk.Chunk.func_177424_a(Chunk.java:776) ~[ase.class:?]
> at net.minecraft.world.World.func_175625_s(World.java:2380) ~[aht.class:?]
> ...

in log when player in this area, continue checking and hope that's stable condition.
